platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/management/internal/autoapply/DefaultAutoAppliedPluginHandler.java

```diff
@@ -104,4 +116,21 @@ private static boolean isAlreadyRequestedInBuildScriptBlock(PluginRequestInterna
     private static boolean hasMatchingCoordinates(Dependency dependency, ModuleIdentifier module) {
         return module.getGroup().equals(dependency.getGroup()) && module.getName().equals(dependency.getName());
     }
+
+    private static Optional<PluginRequestInternal> alternativeCoordinates(PluginRequestInternal plugin) {
```

Comment:
💬 Does it make sense to move it somewhere else? Like `AutoAppliedPluginRegistry`? On the other hand, this seems like a good place to combine all knowledge about how plugins should be excluded from auto-application.

Reply:
It seems to me that this place is too generic. Here, we shouldn't know anything about the develocity vs. gradle enterprise coordinates. This somehow would need to be baked into the auto-applied plugin configuration.

I am still wondering if we can't solve the problem by having the right Gradle Module Metadata on the new develocity plugin - aka it would tell that it replaces the old artifact. Not sure what to do about the plugin id.

Reply:
That's how I would do that with the current approach: https://github.com/gradle/gradle/pull/27365/commits/232ce28c1beb4ef41f2b219342ee57c0e60f86dc

Reply:
As discussed offline, this approach will still be required when we start auto-applying Develocity plugin, as it will need to check for the presence of GE plugin and abort auto-application if it is present.

We also discussed that I will look into what happens now without this change and whether we need to do anything extra for older Gradle versions or Build Scan plugin.